drivers/shared/executor/executor_linux.go

```diff
@@ -145,11 +144,25 @@ func (l *LibcontainerExecutor) cleanOldProcessesInCGroup(nomadRelativePath strin
 			continue
 		}
 
-		err := syscall.Kill(pid, syscall.SIGKILL)
-		if err != nil {
-			l.logger.Error("unable to send signal to process", "pid", pid, "error", err)
+		if err := syscall.Kill(pid, syscall.SIGKILL); err != nil {
+			return fmt.Errorf("unable to send signal to process %d: %v", pid, err)
+		}
+	}
+
+	// Make sure the PID was removed from the cgroup file, otherwise
+	// libcontainer will not be able to launch. Five retries every 100 ms should be
+	// more than enough.
+	for i := 100; i < 501; i += 100 {
+		orphanedPIDs, _ = cgroups.GetAllPids(filepath.Join(root, nomadRelativePath))
+		if len(orphanedPIDs) > 0 {
+			time.Sleep(time.Duration(i) * time.Millisecond)
+			continue
+		} else {
+			l.logger.Info("all orphaned processes killed and removed from cgroup PID file")
```

Comment:
This feels like a noisy log... maybe helpful during development but I'm not sure it's all that useful for operators because we'll get the error log if we don't `return nil`.